### ISSUE TITLE
`listAvailablePackages` returns `No packages were found`

### ISSUE THREAD

**Issue by majguo** [WASdev/azure.websphere-traditional.cluster#56]
Today when I'm testing the latest tWAS cluster testing offer using Ed's entitled IBMid, it always failed. I further triaged the issue and found that the command used to list available packages doesn't run as expected, which caused the entitlement check failed:
```
userName=<your.IBMid>
password=<your.IBMid.pwd>
${IM_INSTALL_DIRECTORY}/eclipse/tools/imutilsc saveCredential -secureStorageFile storage_file -userName "$userName" -userPassword "$password" -passportAdvantage
${IM_INSTALL_DIRECTORY}/eclipse/tools/imcl listAvailablePackages -cPA -secureStorageFile storage_file
```
Command output for listing available packages:
```
No packages were found.
```
I run the above command both in a vm deployed on Azure and my local laptop, with entitled and un-entitled IBMids, but get the same result.
Actually when I re-thought the issue, the similar phenomenon happened few times before, but then disappeared. However, the issue seems persist this time, not sure if the back-end service for listing available packages is down or there is something else wrong.

This issue needs to be triaged and resolved ASAP as it blocks successful tWAS cluster deployments.

**Comment by edburns** [comment]
I recently ran into this again.

We must make it so that such a cause of failure does not show up as "un-entitled".

**Comment by majguo** [comment]
@edburns @git4rk  Re-open this issue to add the check against the result of `listAvailablePackages`:
* display the appropriate error message to the end user if `No packages were found` is returned, which indicates the back-end service for listing available packages is down

**Comment by majguo** [comment]
@gcharters @git4rk Currently, if `listAvailablePackages` returns `No packages were found.`, the user's IBMid will be identified as *un-entitled*, which is inaccurate. As a result, I ask your help to provide the appropriate error message for this case, so that user can clearly understand the deployment failure is due to the unavailability of `listAvailablePackages` used for checking entitlement.

**Comment by gcharters** [comment]
@majguo is there another action after `listAvailablePackages` that acts as the entitlement check?  If so, then maybe something like this as an error message:
`No WebSphere Application Server installation packages were found.  This is likely due to a temporary issue with the installation repository.  Please try again and raise and IBM support issue if the problem persists.`

**Comment by majguo** [comment]
@gcharters Yes. Below is part of output of `listAvailablePackages` for your reference:
```
...
com.ibm.websphere.ND.v90_9.0.5006.20201109_1605
com.ibm.websphere.ND.v90_9.0.5007.20210301_1241
...
com.ibm.websphere.WCT.v90_9.0.5006.20201109_1605
com.ibm.websphere.WCT.v90_9.0.5007.20210301_1241
...
com.ibm.websphere.PLG.v90_9.0.5006.20201109_1605
com.ibm.websphere.PLG.v90_9.0.5007.20210301_1241
...
com.ibm.websphere.IHS.v90_9.0.5006.20201109_1605
com.ibm.websphere.IHS.v90_9.0.5007.20210301_1241
...
com.ibm.java.jdk.v8_8.0.6030.20210512_0631
com.ibm.java.jdk.v8_8.0.6031.20210513_1113
...
```
Based on @git4rk 's response before, `ND.v90_9.0.5007` is the first entitled version which is only accessible by the entitled IBM ID. In other words, no `ND.v90_9.0.5007` is returned in the output of `ND.v90_9.0.5007` for an un-entitled IBM ID. So the next action after `listAvailablePackages` is to check whether the output includes literal string `ND.v90_9.0.5007`.

**Comment by git4rk** [comment]
@majguo @gcharters Here is the suggested text from the IDD team (Pam):
No WebSphere Application Server installation packages were found. This is likely due to a temporary issue with the installation repository. Try again and open an IBM Support issue if the problem persists.

**Comment by git4rk** [comment]
From 2021-07-06 meeting minutes:
- listAvailablePackages returns No packages were found: related issue #39 (see above)
    - Option 1: problem with service.  Returns "No packages were found"
    - Option 2: not entitled. Returns a set of packages, but not the one we're looking for.
    - Option 3: entitled. Returns a set of packages including the one we were looking for.
- Action: Graham Investigate the issues with listAvailablePackages issue on IBM side.  Why it's happening quite frequently and lasting for up to a day.

**Comment by git4rk** [comment]
Posted to IM Forum: https://w3.ibm.com/w3publisher/capilano-installation-manager-etc/forum#/discussion/110/imcl-listavailablepackages-cpa-fails-with-no-packages-were-found

**Comment by edburns** [comment]
This issue can be closed.  If the problem resurfaces, we will open another one.  @git4rk .

**Comment by majguo** [comment]
Closing the issue as it's fixed in [Create GitHub Actions workflow to verify if entitlement check works WASdev/azure.websphere-traditional.image#57](https://github.com/WASdev/azure.websphere-traditional.image/pull/57).
